src/jit/codegenarm64.cpp

```diff
@@ -3518,53 +3518,57 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             emit->emitIns_R_L(INS_adr, EA_PTRSIZE, genPendingCallLabel, targetReg);
             break;
 
-        case GT_COPYOBJ:
-            genCodeForCpObj(treeNode->AsCpObj());
+        case GT_STORE_OBJ:
+            if (treeNode->OperIsCopyBlkOp())
+            {
+                assert(treeNode->AsObj()->gtGcPtrCount != 0);
+                genCodeForCpObj(treeNode->AsObj());
+            }
             break;
+            __fallthrough;
```

Comment:
No - and it's a bit surprising that this wasn't discovered, though I think most of the no-gc-ptr copies get a GT_BLK on the lhs from the beginning.

---
In reply to: [77274977](https://github.com/dotnet/coreclr/pull/7027#discussion_r77274977) [](ancestors = 77274977)